### PR TITLE
feat: Add agoric-upgrade-17 (proposal 78)

### PR DESCRIPTION
**Pull request by gibson042**
Created by following
https://github.com/Agoric/agoric-sdk/tree/agoric-upgrade-17/a3p-integration/proposals#migrate-the-proposal for https://ping.pub/agoric/gov/78

This PR currently copies `a:upgrade-17` but _not_ `s:stake-bld` or `z:acceptance`, which per [CONTRIBUTING.md](https://github.com/Agoric/agoric-3-proposals/blob/main/CONTRIBUTING.md) I think should instead move to a3p-integration/z:acceptance.

I admit to a lack of understanding in how the transfer for this upgrade should proceed, and will update instructions based on review feedback.

### CODE REVIEW

File: proposals/78:upgrade-17/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Generated Artifacts to Ignore in CI
+add-LEMONS/
+add-OLIVES/
+upgrade-bank/
+upgrade-provisionPool/
+upgrade-orch-core/
```
Comment (dckc) [inline proposals/78:upgrade-17/.gitignore:6]: I think these are generated in the agoric-sdk context but in this context, they're checked in.

File: proposals/78:upgrade-17/agd-tools.js [hunk 2]
```diff
@@ -0,0 +1,208 @@
+import {
```
Comment (dckc) [inline proposals/78:upgrade-17/agd-tools.js:1]: sigh. another copy of these tools. Here's hoping to migrate stuff to synthetic-chain

File: proposals/78:upgrade-17/README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Proposal to upgrade the chain software
+
+This reflects agoric-upgrade-17, which was adopted on mainnet by [governance
+proposal 78](https://ping.pub/agoric/gov/78).
```
Comment (dckc) [inline proposals/78:upgrade-17/README.md:4]: maybe include the date? it passed 2024-10-05

File: proposals/78:upgrade-17/localchain.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import test from 'ava';
+
+import { agd, evalBundles, waitForBlock } from '@agoric/synthetic-chain';
+
+const SUBMISSION_DIR = 'localchaintest-submission';
```
Comment (dckc) [inline proposals/78:upgrade-17/localchain.test.js:5]: I think we need this directory populated with files that we built with `a3p-integration/scripts/build-all-submissions.sh`.

File: proposals/78:upgrade-17/provisionPool.test.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  getIncarnation,
+  waitForBlock,
+} from '@agoric/synthetic-chain';
+
+import { bankSend, getProvisionPoolMetrics } from './agd-tools.js';
+
+const NULL_UPGRADE_BANK_DIR = 'upgrade-bank';
+const UPGRADE_PP_DIR = 'upgrade-provisionPool';
+const ADD_LEMONS_DIR = 'add-LEMONS';
+const ADD_OLIVES_DIR = 'add-OLIVES';
```
Comment (dckc) [inline proposals/78:upgrade-17/provisionPool.test.js:16]: likewise I think we need these directories populated.

File: proposals/78:upgrade-17/synthetic-chain-excerpt.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
+/**
+ * @file work-around: importing @agoric/synthetic-chain hangs XXX
```
Comment (dckc) [inline proposals/78:upgrade-17/synthetic-chain-excerpt.js:2]: ??? I wonder what that's about.

But I guess it passed review in agoric-sdk, so... oh well.

**Review by dckc — COMMENTED** [review]
close! but due to missing stuff from `build-all-submissions.sh`, I doubt ci will pass.

File: proposals/78:upgrade-17/localchain.test.js (hunk 4 above)
Comment (gibson042) [reply proposals/78:upgrade-17/localchain.test.js:5]: Confirmed; CI got some expected errors:
```
PPT post {
  totalMintedConverted: { brand: '$0.Alleged: IST brand', value: '+20010000' },
  totalMintedProvided: { brand: '$0', value: '+2750000' },
  walletsProvisioned: '+11'
}
  ✘ [fail]: provisionPool › upgrading provisionPool and vat-bank Rejected promise returned by test
    ℹ add assets before
    ℹ upgrade Bank
  ─

  localchain › localchain passes tests

  Rejected promise returned by test. Reason:

  Error {
    code: 'ENOENT',
    errno: -2,
    path: 'localchaintest-submission',
    syscall: 'scandir',
    message: 'ENOENT: no such file or directory, scandir \'localchaintest-submission\'',
  }



  provisionPool › upgrading provisionPool and vat-bank
  Rejected promise returned by test. Reason:

  Error {
    code: 'ENOENT',
    errno: -2,
    path: 'upgrade-bank',
    syscall: 'scandir',
    message: 'ENOENT: no such file or directory, scandir \'upgrade-bank\'',
  }
```

**Review by dckc — APPROVED** [review]
well done.

¡bien hecho!

**Pull request merged**